File: src/Services/Course/Course.Api/Models/Teacher.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Course.Api.Models
{
    public class Teacher
    {
        public int Id { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        // Should be abbreviation?
        public string Initials { get; set; }
        public string Email { get; set; }
        public string Phone { get; set; }

        public List<Lecturer> Lecturers { get; set; }
    }
}

File: src/Web/Spa/src/app/models/teacher.ts
import { School } from './school';
import { Room } from './room';
import { Building } from './building';

export class Teacher {
    id:number;
    pcn:string;
    firstName:string;
    lastName:string;
    initials:string;
    email:string;
    phoneNumber:string;
    accountStatus:string;
    school:School;
    room:Room;
    building:Building;
}